client/src/components/SearchResultCard.vue

```diff
@@ -149,7 +149,11 @@ export default {
 			this.$router.push(`/data-sources/${this.dataSource.airtable_uid}`);
 		},
 		openSource() {
-			window.open(this.dataSource.source_url, '_blank');
+			let url = this.dataSource.source_url;
+			if (!/^https?:\/\//i.test(url)) {
+				url = 'https://' + url;
+			}
+			window.open(url, '_blank');
```

Comment:
The regex pattern string here is going to be hard for someone not knowledgeable about regex/this code base/both to know what's going on.

In this case, I recommend doing one of two things:
1. Leaving a comment describing why this code is here and what it does
2. Extracting the complicated code to a separate method with a clear function name.

2 is my preference here, because this code is already heavily indented and such heavy indents can make the code more difficult to parse.

Reply:
For example, a method named `prepend_url_prefix_if_not_found()`